doc/adrs/031_separate_metadata_from_representations.adoc

```diff
@@ -0,0 +1,101 @@
+For that, we will introduce an interface named `IRepresentationMetadata` with the following content in the schema:
+
+```
+interface RepresentationMetadata {
+  id: ID!
+  descriptionId: ID!
+  label: String!
+}
+```
+
+With this interface, we wil have various implementations (one for each representation) such as:
+
+```
+type DiagramMetadata implements RepresentationMetadata {
+  id: ID!
+  descriptionId: ID!
+  label: String
+  toolSections: [ToolSection!]!
+}
+```
+
+or
+
+```
+type FormMetadata implements RepresentationMetadata {
+  id: ID!
+  descriptionId: ID!
+  label: String
+}
+```
+
+Everytime a GraphQL API allowed to retrieve a representation except in the subscription (with the various refresh event payload), we will provide the metadata instead.
+As a result, the following fields will be modified:
+
+- EditingContext#representation
+- EditingContext#representations
+- CreateRepresentationSuccessPayload#representation
+- DeleteFromDiagramSuccessPayload#diagram
+- EditLabelSuccessPayload#diagram
+- InvokeNodeToolOnDiagramSuccessPayload#diagram
+- InvokeEdgeToolOnDiagramSuccessPayload#diagram
+
+Most of those fields are not very useful and they may be removed altogether from the GraphQL API.
+This change does not require any change in our Java data structure, it will just make it impossible with GraphQL to retrieve the content of a representation outside of its subscription.
+
+=== Representation change
+
+We will introduce a new concept named `IRepresentationMetadata` which will contain identity-related properties while the `IRepresentation` will keep the content.
+We will remove those properties from the various representations to move them to `IRepresentationMetada`:
+
+- id
+- descriptionId
+- label
+- kind
+
+We will also introduce the concept of `ISemanticRepresentationMetadata` in a similar fashion as the `ISemanticRepresentation` with the following properties:
+
+- targetObjectId
+
+The representations and their descriptions will be updated to remove those fields, which will include changes to:
+
+- Diagram
+- Tree
+- Selection
+- Form
+- Validation
+
+The various renderer will not have to handle those properties anymore.
+We will also be able to delete the `GetOrCreateRandomIdProvider` since we won't need those kind of behavior to keep a stable identifier.
+We will have the ability to remove `DiagramDescription.LABEL` too.
+
+To retrieve said representation metadata if we need it which is not sure in Sirius Components, we will probably add an `IRepresentationMetadataSearchService`.
+Implementation of the IRepresentationMetadata will be left to the description of the server using Sirius Component.
+The kind will be used to define the type of the representation metadata in the GraphQL API.
+This field will not appear in the GraphQL API.
+There won't be any DiagramMetadata or FormMetadata Java class in Sirius Components.
+
+== Status
+
+WIP
```

Comment:
It's more of an open question than a proposal, but could we have GraphQL fields like `DiagramMetadata#content`? From my understanding it would allow the client to "dig into" the actual content if needed everywhere it has a `DiagramMetadata` while still decoupling the two parts (and not actually loading the diagram's content unless the query explicitly asks for it).

Reply:
I already reply to this comment privately but I'll reply here too in order to have all comments here.

Currently, the only way we are really using the representation content is thanks to the subscription. So right now, I would gladly add a field `Representation#metadata: RepresentationMetadata" but since we do not need the representation anywhere else, I would be opposed to adding a field like `RepresentationMetadata#content: Representation` since it would open a powerful API that I would like to contain for now.

Technically resolving `RepresentationMetadata#content: Representation` should require us to load the representation but should we retrieve it from the datastore or from the RepresentationEventProcessor? Should we load the whole EditingContext if the representation is not opened right now? There are solutions to those questions but they come with serious tradeoffs so we would need to make a real choice which is not needed right now.

Reply:
OK, I understand.